zio-schema/shared/src/main/scala/zio/schema/DynamicValue.scala

```diff
@@ -110,12 +123,26 @@ sealed trait DynamicValue {
       case (dyn, Schema.Dynamic(_)) => Right(dyn)
 
       case _ =>
-        Left(() => s"Failed to cast $self to schema $schema")
+        Left(DynamicValue.Err.CastError(self, schema))
     }
 
 }
 
 object DynamicValue {
+
+  sealed trait Err
```

Comment:
Let's call this `Error` instead of `Err` to mirror `zio.Config.Error`.

Reply:
There is already an Error type defined inside DynamicValue

Reply:
🤦‍♂️ 

`DecodeError`, maybe?